dbt-athena/src/dbt/adapters/athena/impl.py

```diff
@@ -1173,6 +1177,20 @@ def delete_from_glue_catalog(self, relation: AthenaRelation) -> None:
                 LOGGER.error(e)
                 raise e
 
+    @available
+    def drop_glue_database(self, database_name: str, catalog_id: Optional[str] = None) -> None:
+        conn = self.connections.get_thread_connection()
+        creds = conn.credentials
+        client = conn.handle
+
```

Comment:
in case the user it's using setups where the account it's different than the current account we might need to use:
`catalog_id = get_catalog_id(data_catalog)` (see how we do for the [get_table](https://github.com/dbt-labs/dbt-athena/blob/1947ebbef5f77c24c63748d1f2e96bd88b8a363b/dbt-athena/src/dbt/adapters/athena/impl.py#L340)) and pass it to the `delete_database` call.

What do you think about that? I believe that it's a pretty edge case, but better to add it now than later.

Reply:
hi @nicor88 😄 definitely, you are so fast
thx for pointing that our, I've faced this issue on my local test setup and updated the logic, pls review as now it seems to be ready

Reply:
Nice, looks good :)